### ISSUE TITLE
Review ShellNavigationItem: Model or ViewModel?

### ISSUE THREAD

**Issue by Alvaromah** [Microsoft/WindowsTemplateStudio#1283]
ShellNavigationItem acts more like a Model than a ViewModel.

If we agree that, it should inherit from ObservableObject, not ViewModelBase, and be located at the "Models" folder.

**Comment by ralarcon** [comment]
We need to carefully evaluate the impact that a change like this will introduce in existing apps. We'll review.

**Comment by crutkas** [comment]
Can you explain more why you think that is.

**Comment by ralarcon** [comment]
I guess you are asking @Alvaromah, isn't it?

**Comment by crutkas** [comment]
@ralarcon.  Yes

**Comment by Alvaromah** [comment]
While it is not definitely clear when to use a ViewModel or a Model, in my opinion, a ViewModel must be responsible for handling the View logic and the Model acts more like a data wrapper, helping its representation in the View.

In this case, the ShellViewModel is handling the navigation logic, while the ShellNavigationItem wraps the definition (data) of what a MenuItem is: an Icon, a Label and a action when is selected (i.e: a page to navigate to).

On the other hand, current design implements always a pair View-ViewModel, except for the ShellNavigationItem. Furthermore, the name convention for a ViewModel is always “<view name>ViewModel”, but not for the ShellNavigationItem.

In summary, I see a NavigationItem more like a data representation than a View representation :)

**Comment by mrlacey** [comment]
The `ShellNavigationItem` (SNI) could be interpreted as part of either the `Model` or the `ViewModel` layer. It's in that gray area where different people will argue for both given different preferences, past experiences, and how they define the limits of what should be in each layer.

Let me start with some simplified definitions that will help explain why it's where it is.
**View** - The visual representation of the app.
**Model** - The core business data, rules, and logic used by the app.
**ViewModel** - The bridge between the View and Model.

Given these definitions, it is clear that the SNI is not tied to the model. It exists purely as part of the definition of what should appear in the UI and provides 

Some people claim that the ViewModel is the data and logic related to the UI. Therefore, as the SNI is only data it can't be in the VM layer. This is to be overly strict in a definition and ignores that the data in the SNI powers the Shell. Yes, the SNI is a POCO containing no specific logic but the data it holds is specific to the UI.

Another way to consider the definition of the different layers is in terms of their reuse when building for different platforms or different UIs. (Ignore the multi-platform aspect of a UWP app for now as I'm thinking about different devices/platforms requiring different apps.)
**Model** - Has logic that is platform independent.
**View** - Will normally be unique to an individual app.
**ViewModel** - May have some reuse but is more likely to be tightly coupled to the View.

If you were to completely change the UI and not need the SNI as part of the top level navigation, this should only impact the View and ViewModel layers. The Model could remain unchanged.

WTS does this essentially with the different project types. If you created a Navigation Pane or Tabbed version of a project, the only difference is in the View and VM layers.

Separation of logic and the ability to reuse code in different layers is also a good indication of the layer something should be in. It's ok for the View to depend on the VM but it should not depend on the Model directly. The VM should depend on only the Model. The Model should not depend on anything in either of the other layers. 
The SNI contains platform specific items. Specifically, it has references to `Windows.UI.Xaml` which means it is tightly coupled to the UWP View layer and could not be reused in other code like a good Model layer should. If we were to move the SNI to the Model layer we should really also remove these dependencies. Doing so would bring little tangible benefit to the code as it is currently generated though.
_Note. I acknowledge, the current WTS output does not separate the layers well as it puts everything in a single assembly, but this will change in a future update._


If you think it would be better as a Model in your code base you can always move it in your code but I do not currently think it is the best place for it.
If we were to move it we would need a very good reason.
